docs/guide-kms.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Using KMS Proxy"
+layout: docs-enclaver-single
+category: guides
+---
+
+# KMS Proxy Overview
+
+To derive the most value from Enclaver, your application's sensitive data should be encrypted using a key stored in AWS Key Management Service (KMS) to be decrypted only within an enclave.
+To ensure the data can only be decrypted from within an enclave running a specific image hash, a KMS key policy can constrain the Decrypt operation to requests with given PCR values (hashes).
+See [Using cryptographic attestation with AWS KMS](https://docs.aws.amazon.com/enclaves/latest/user/kms.html) for details.
+The PCR values are included in an attestation document, signed by AWS. An application running inside an enclave can retrieve such a document and attach it to the KMS requests.
+However it is a cumbersome process to be done directly by the application. The Enclaver includes a KMS proxy that makes this process transparent.
+By configuring the application to connect to the KMS proxy running on `localhost`, all outgoing KMS requests that support it (e.g. Decrypt) will have the attestation documents attached to them.
+
+# Enabling the KMS Proxy
+
+Enable the KMS proxy in your `enclaver.yaml` as follows:
+
+```yaml
+kms_proxy:
+  listen_port: 9999
+
+egress:
+  allow:
+    - 169.254.169.254
+    - kms.*.amazonaws.com
+```
+
+When the enclave starts up, Enclaver will define a `AWS_KMS_ENDPOINT=http://127.0.0.1:9999` environment variable. The value can be passed into the AWS SDK to override the default endpoint.
+The exact details are language specfiic. See below for examples of the most popular languages.
+
+*WARNING*: Do not expose the KMS proxy port (9999 in this example) in the `ingress` section. Doing so will expose the KMS proxy outside the enclave, allowing untrusted code to decrypt the data.
+
+# Configuring AWS SDK with KMS Proxy Endpoint
+
+<details>
```

Comment:
I don't think you need to be worried about the length of showing each of the code snippets.